django/db/models/functions/math.py

```diff
@@ -0,0 +1,171 @@
+import math
+
+from django.db.models import FloatField, Func, Transform
+
+
+class Abs(Transform):
+    function = 'ABS'
+    lookup_name = 'abs'
+
+
+class ACos(Transform):
+    function = 'ACOS'
+    lookup_name = 'acos'
+    output_field = FloatField()
+
+
+class ASin(Transform):
+    function = 'ASIN'
+    lookup_name = 'asin'
+    output_field = FloatField()
+
+
+class ATan(Transform):
+    function = 'ATAN'
+    lookup_name = 'atan'
+    output_field = FloatField()
+
+
+class ATan2(Func):
+    function = 'ATAN2'
+    arity = 2
+    output_field = FloatField()
+
+
+class Ceil(Transform):
+    function = 'CEILING'
+    lookup_name = 'ceil'
+
+    def as_oracle(self, compiler, connection):
+        return super().as_sql(compiler, connection, function='CEIL')
+
+
+class Cos(Transform):
+    function = 'COS'
+    lookup_name = 'cos'
+    output_field = FloatField()
+
+
+class Cot(Transform):
+    function = 'COT'
+    lookup_name = 'cot'
+    output_field = FloatField()
+
+    def __init__(self, num):
+        super().__init__(num)
+
+    def get_tan(self):
+        return Tan(self.source_expressions[0])
+
+    def as_oracle(self, compiler, connection):
+        cot = 1 / self.get_tan()
+        return float(cot)
```

Comment:
@JunyiJ My previous suggestion was to use the `TAN` database function on Oracle, i.e. 
```python
def as_oracle(self, compiler, connection):
    return super().as_sql(compiler, connection, template='1 / TAN(%(expressions)s)')
```

Reply:
I see. Thanks @felixxm. I have to admit that I am not so familiar with the uses of template and as_sqal(...). I updated the Cot already. Do you mind to help me check the other oracle functions: Degrees, Pi and Radians in the directory: django/db/models/functions/math.py ? Testing on oracle is too slow (generally 1 hour to run the oracle test).